rdwatch/core/models/model_run.py

```diff
@@ -54,5 +57,96 @@ class ProposalStatus(models.TextChoices):
+        blank=True,
+        null=True,
+        help_text='The latest timestamp of any evaluation for this model run',
+    )
+    cached_bbox = PolygonField(
+        srid=4326,
+        blank=True,
+        null=True,
+        help_text='The bounding box of all evaluations for this model run',
+        spatial_index=False,
+    )
+
     def __str__(self) -> str:
         return str(self.pk)
+
+    def compute_aggregate_stats(self):
+        """
+        Compute denormalized aggregate stats and persist them on this model run.
+
+        Many of the fields we report for each model run are expensive to compute, and need to be
+        returned to the client often. Calling this will perform the expensive computations and
+        persist the results on the model run for fast retrieval.
+
+        This should be called whenever a model run is fully ingested, and again whenever any of the
+        underlying data corresponding to the model changes such that these summary statistics might
+        be affected.
+        """
+        stats = ModelRun.objects.filter(pk=self.pk).aggregate(
+            cached_bbox=BoundingBoxPolygon('evaluations__geom'),
+            cached_numsites=models.Count('evaluations__pk', distinct=True),
+            cached_score=models.Avg('evaluations__score'),
+            cached_timerange_min=models.Min('evaluations__start_date'),
+            cached_timerange_max=models.Max('evaluations__end_date'),
+            cached_timestamp=models.Max('evaluations__timestamp'),
+        )
+        self.cached_bbox = stats['cached_bbox']
+        self.cached_numsites = stats['cached_numsites']
+        self.cached_score = stats['cached_score']
+        self.cached_timerange_min = stats['cached_timerange_min']
+        self.cached_timerange_max = stats['cached_timerange_max']
+        self.cached_timestamp = stats['cached_timestamp']
+
+        self.save(
+            update_fields=[
+                'cached_bbox',
+                'cached_numsites',
+                'cached_score',
+                'cached_timerange_min',
+                'cached_timerange_max',
+                'cached_timestamp',
+            ]
+        )
+
+    @classmethod
+    def compute_all_aggregate_stats(cls, recompute_all: bool = False):
+        """
+        Computes aggregate stats on all model runs that are missing them.
+
+        If `recompute_all` is True, all model runs will have their aggregate stats recomputed
+        regardless of whether they are already populated.
+        """
+        if recompute_all:
+            qs = cls.objects.all()
+        else:
+            qs = cls.objects.filter(cached_numsites=None)
+
+        # Unfortunately, Django ORM doesn't support joins in update queries, so doing this in a
+        # single join query would require raw SQL that would be a pain to maintain.
+        # Since this should only rarely run, we don't mind looping here.
+        for model_run in qs:
+            model_run.compute_aggregate_stats()
```

Comment:
The number of model runs in the system could be significant. I believe the T&E deployment has tens of thousands of them. I'm not sure if it's better to use `.iterator()` here to avoid loading all that data into memory at once (at the cost of speed), or if that scale of data is still okay to do a regular iteration over, any thoughts?